pep-0566.rst

```diff
@@ -15,12 +13,23 @@ Post-History:
 Replaces: 345
 Resolution: https://mail.python.org/pipermail/distutils-sig/2018-February/032014.html
 
+
+.. important::
+   This PEP is a historical change proposal; the up-to-date, canonical version
+   of the specification it modifies is
+   `maintained on the PyPA Specifications section of packaging.python.org
+   <https://packaging.python.org/en/latest/specifications/core-metadata/>`__.
+   See the `PyPA specification update process
+   <https://www.pypa.io/en/latest/specifications/#handling-fixes-and-other-minor-updates>`__
+   for how to propose editorial and content changes.
+
+
```

Comment:
I'm concerned that there's an inconsistency here - you're suggesting that the Metadata 1.2 spec is marked as Superseded, because of this spec, but not that this spec is Superseded by the Metadata 2.2 spec.

Also, Metadata 2.1 is definitely not dead - many tools still generate it. We do encourage tools to migrate to the latest metadata version, but this is often a slow process, and older files (such as wheels for previous versions) may never change. It's superseded for *writers* of metadata, but not for *readers*. (Specifically, the Metadata 2.2 spec does *not* say that readers can reject a metadata version of 2.1, and similarly for other versions).

So while I agree that marking metadata versions as superseded whenever a new version comes out is *technically* correct, it's extra work for little practical benefit, and it's arguably misleading in practice. Is it not sufficient to note that the official documentation is elsewhere? Core language PEPs frequently change things that were previously defined by an earlier PEP, and we don't try to document all the implications of that in the PEP headers.

Reply:
Well, "historical" doesn't mean "dead", no more than "Final" does...but if there is confusion about that, we can certainly use a different term or elide that phrase entirely. PEP 566 is a bit of a special case since it marks a transition point between the metadata PEPs being full specifications (like Metadata 1.2) and just change proposals with an atomic set of changes to the canonical spec, with the bump to the Metadata version being secondary (like PEP 643). It still specifies that it is a new metadata version itself rather than putting the focus on proposed change(s), but it also only specifies the changes versus 1.2 rather than the full spec.

> So while I agree that marking metadata versions as superseded whenever a new version comes out is technically correct, it's extra work for little practical benefit, and it's arguably misleading in practice. 

Sorry I didn't make it clear—my thinking here was to not mark it or any later PEPs framed as changed proposals as "superseded", just Final and with the admonition linking to the up to date spec once they are implemented in tools, so there wouldn't be any further maintenance overhead after this PR since the admonition would take care of that for us. The grounds for the older complete specs being marked as superseded was that they specify complete metadata versions and there are newer metadata versions available, which marking them as superseded makes clear and allows linking and easy navigation to the previous and next, but this does not mean they are invalid (Rejected/Withdrawn).